java/client/src/org/openqa/selenium/devtools/target/Target.java

```diff
@@ -0,0 +1,311 @@
+   * beforeunload hooks.EXPERIMENTAL
+   */
+  @Beta
+  public static Command<Void> disposeBrowserContext(BrowserContextID browserContextID) {
+    Objects.requireNonNull(browserContextID, "browserContextId is required");
+    return new Command<>(
+        "Target.disposeBrowserContext", ImmutableMap.of("browserContextId", browserContextID));
+  }
+
+  /**
+   * Returns information about a target.EXPERIMENTAL
+   */
+  @Beta
+  public static Command<TargetInfo> getTargetInfo(Optional<TargetId> targetId) {
+    final ImmutableMap.Builder<String, Object> params = ImmutableMap.builder();
+    targetId.ifPresent(targetID -> params.put("targetId", targetID));
+    return new Command<>(
+        "Target.getTargetInfo",
+        params.build(),
+        ConverterFunctions.map("targetInfo", TargetInfo.class));
+  }
+
+  /**
+   * Retrieves a list of available targets.
+   */
+  public static Command<List<TargetInfo>> getTargets() {
+    return new Command<>(
+        "Target.getTargets",
+        ImmutableMap.of(),
+        ConverterFunctions.map("targetInfos", new TypeToken<List<TargetInfo>>() {
+        }.getType()));
+  }
+
+  /**
+   * Controls whether to discover available targets and notify via
+   * targetCreated/targetInfoChanged/targetDestroyed events.
+   */
+  public static Command<Void> sendMessageToTarget(
+      String message, Optional<SessionId> sessionID, @Deprecated Optional<TargetId> targetID) {
+    Objects.requireNonNull(message, "message is required");
+    final ImmutableMap.Builder<String, Object> params = ImmutableMap.builder();
+    params.put("message", message);
+    sessionID.ifPresent(sessionId -> params.put("sessionId", sessionId));
+    targetID.ifPresent(targetId -> params.put("targetId", targetId));
+    return new Command<>("Target.sendMessageToTarget", params.build());
+  }
+
+  /**
+   * Attaches to the target with given id.
+   *
+   * @return {@link SessionId}
+   */
+  public static Command<SessionId> attachToTarget(TargetId targetId, Optional<Boolean> flatten) {
+    Objects.requireNonNull(targetId, "Target ID must be set.");
+    final ImmutableMap.Builder<String, Object> params = ImmutableMap.builder();
+    params.put("targetId", targetId);
+    params.put("flatten", flatten.orElse(true));
+//    flatten.ifPresent(aBoolean -> params.put("flatten", aBoolean));
+    return new Command<>(
+        "Target.attachToTarget",
+        params.build(),
+        ConverterFunctions.map("sessionId", SessionId.class));
+  }
+
+  /**
+   * Controls whether to automatically attach to new targets which are considered to be related to
+   * this one. When turned on, attaches to all existing related targets as well. When turned off,
+   * automatically detaches from all currently attached targets.EXPERIMENTAL
+   */
+  public static Command<Void> setAutoAttach(
+      Boolean autoAttach, Boolean waitForDebuggerOnStart, Optional<Boolean> flatten) {
```

Comment:
We can clean this up in a later pass, but we can use primitives for parameters that are required. `boolean`, as we do elsewhere.